src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3WebtransportSession.cs

```diff
@@ -0,0 +1,184 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Net.Quic;
+using System.Runtime.Versioning;
+using System.Text;
+using System.Net.Http.Headers;
+using System.Net.Security;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Runtime.CompilerServices;
+using System.IO;
+using System.Threading.Channels;
+
+namespace System.Net.Http
+{
+    [SupportedOSPlatform("windows")]
+    [SupportedOSPlatform("linux")]
+    [SupportedOSPlatform("macos")]
+    [UnsupportedOSPlatform("browser")]
+
+    public class Http3WebtransportSession : IAsyncDisposable, IDisposable
+    {
+        private readonly QuicStream _connectStream;
+        private static HttpMessageInvoker? s_defaultInvoker;
+
+        public long Id
+        {
+            get { return _connectStream.Id; }
+        }
+
+        private readonly Channel<QuicStream> _incomingStreamsQueue = Channel.CreateUnbounded<QuicStream>(new UnboundedChannelOptions()
+        {
+            SingleWriter = true
+        });
+
+        private int _disposed;
+        internal Http3WebtransportManager _WebtransportManager;
+
+        internal const string WebTransportProtocolValue = "webtransport";
+        internal const string VersionEnabledIndicator = "1";
+        internal const string SecPrefix = "sec-webtransport-http3-";
+        internal const string VersionHeaderPrefix = $"{SecPrefix}draft";
+        internal const string CurrentSuppportedVersion = $"{VersionHeaderPrefix}02";
+
+        internal Http3WebtransportSession(QuicStream connectStream, Http3WebtransportManager manager)
+        {
+            _WebtransportManager = manager;
+            _connectStream = connectStream;
+            _ = _connectStream.WritesClosed.ContinueWith(async t =>
+            {
+                await DisposeAsync().ConfigureAwait(false);
+
+            }, CancellationToken.None, TaskContinuationOptions.ExecuteSynchronously, TaskScheduler.Current);
+
+        }
+
+        /// <summary>
+        /// Creates a webtransport session by creating a webtransport connect request, sending it to <see cref="Uri">uri</see>.
+        /// </summary>
+        public static async ValueTask<Http3WebtransportSession?> ConnectAsync(Uri uri, HttpMessageInvoker? invoker, CancellationToken cancellationToken)
```

Comment:
I don't think there are any tests that test this method without passing an invoker (testing scenario with a default invoker)?

Reply:
@CarnaViire As per our last discussion you suggested me to look at websockets. I looked and I do not think we use the default invoker in websockets, in both [Connect.Http2 test for websockets](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Net.WebSockets.Client/tests/ConnectTest.Http2.cs) and [SendReceive.Http2](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Net.WebSockets.Client/tests/SendReceiveTest.Http2.cs) we include in the `Connect Async` function the [`GetInvoker`](https://cs.github.com/dotnet/runtime/blob/de0c8a5500e90f1b63b038bcf7b40dc577529be2/src/libraries/System.Net.WebSockets.Client/tests/ClientWebSocketTestBase.cs#L123) function which creates a new client/invoker
Am I missing something?